lib/storage-handler.js

```diff
@@ -253,6 +253,7 @@ exports.download = function(provider, req, res, container, file, cb) {
     reader.pipe(res);
 
     reader.on('error', function onReaderError(err) {
+      res.set('content-type', 'application/json');
```

Comment:
>my suggestion was to modify the signature of processError, but im not sure if that make sense either

ummm that refactoring of `processError()` could make sense... @bajtos what do you think?